src/views/webview-app/components/connect-form/general-tab/host/host.tsx

```diff
@@ -0,0 +1,352 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlus, faMinus } from '@fortawesome/free-solid-svg-icons';
+
+const styles = require('./host.less');
+
+enum CONNECTION_TYPE {
+  STANDALONE = 'STANDALONE',
+  REPLICA_SET = 'REPLICA_SET',
+  SRV_RECORD = 'SRV_RECORD'
+}
+
+const CONNECTION_TYPE_OPTIONS: {
+  [connectionType in CONNECTION_TYPE]: {
+    label: string;
+    value: CONNECTION_TYPE;
+  }
+} = {
+  [CONNECTION_TYPE.STANDALONE]: {
+    label: 'Standalone',
+    value: CONNECTION_TYPE.STANDALONE
+  },
+  [CONNECTION_TYPE.REPLICA_SET]: {
+    label: 'Replica Set',
+    value: CONNECTION_TYPE.REPLICA_SET
+  },
+  [CONNECTION_TYPE.SRV_RECORD]: {
+    label: 'SRV Record',
+    value: CONNECTION_TYPE.SRV_RECORD
+  }
+};
+
+type StateProps = {
+  hostname: string;
+  hosts: Host[];
+  isSrvRecord: boolean;
+  port: number;
+  replicaSet?: string;
+};
+
+type DispatchProps = {
+  onHostnameChanged: (newHostName: string) => void;
+  onPortChanged: (newPort: number) => void;
+  setReplicaSet: (replicaSet?: string) => void;
+  toggleSRVRecord: () => void;
+  updateHosts: (hosts: Host[]) => void;
+};
+
+export class HostInput extends React.PureComponent<StateProps & DispatchProps> {
+  onConnectionTypeChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    switch (event.target.value as CONNECTION_TYPE) {
+      case CONNECTION_TYPE.STANDALONE:
+        this.props.setReplicaSet();
+        if (this.props.hosts.length > 1) {
+          // Remove extra hosts when moving to standalone from replica set.
+          this.props.updateHosts(this.props.hosts.slice(0, 1));
+        }
+        if (this.props.isSrvRecord) {
+          this.props.toggleSRVRecord();
+        }
+
+        break;
+      case CONNECTION_TYPE.REPLICA_SET:
+        if (this.props.replicaSet === undefined) {
+          this.props.setReplicaSet('');
+        }
+        if (this.props.isSrvRecord) {
+          this.props.toggleSRVRecord();
+        }
+
+        break;
+      case CONNECTION_TYPE.SRV_RECORD:
+        this.props.setReplicaSet();
+
+        if (!this.props.isSrvRecord) {
+          this.props.toggleSRVRecord();
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  onHostChanged = (updatedHost, hostIndex): void => {
+    const { hosts } = this.props;
+
+    const newHosts = [ ...hosts ];
+
+    newHosts[hostIndex] = updatedHost;
+
+    this.props.updateHosts(newHosts);
+  };
+
+  onAddNewHost = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    event.preventDefault();
+
+    const { hosts, replicaSet } = this.props;
+
+    this.props.updateHosts([
+      ...hosts,
+      {
+        ...DEFAULT_HOST
+      }
+    ]);
+
+    if (!replicaSet) {
+      this.props.setReplicaSet('');
+    }
+  };
+
+  onHostnameChanged = (event): void => {
+    this.props.onHostnameChanged(event.target.value);
+  };
+
+  onPortChanged = (event): void => {
+    this.props.onPortChanged(event.target.value.trim());
```

Comment:
```suggestion
    this.props.onPortChanged(+event.target.value.trim());
```

because the typescript definitions say it’s a number? :)

Reply:
nice catch - updated - also added the event types to the function since it was thinking `event` was an `any`.